scala/lambdas/ingest-validate-generic-ingest-inputs/src/main/scala/uk/gov/nationalarchives/ingestvalidategenericingestinputs/Lambda.scala

```diff
@@ -0,0 +1,261 @@
+      entriesWithValidatedUuids = valueValidator.checkIfAllIdsAreUuids(entriesWithValidatedUniqueIds)
+
+      entryTypesGrouped = allEntryIds.groupBy { case (_, entryType) => entryType }
+      entriesWithValidatedParentIds = valueValidator.checkIfEntriesHaveCorrectParentIds(entriesWithValidatedUuids, allEntryIds.toMap, entryTypesGrouped)
+      entriesCompletelyValidated = valueValidator.checkForCircularDependenciesInFolders(entriesWithValidatedParentIds)
+
+      jsonStructuralErrors = {
+        minimumAssetsAndFilesErrors.zip(LazyList.from(1)).map { (error, errorNum) =>
+          s"MinimumAssetsAndFilesError $errorNum:" -> error.swap.toOption.map(nel => InvalidValidationResult(nel.toList)).get
+        } :::
+          atLeastOneEntryWithSeriesAndNullParentErrors.zip(LazyList.from(1)).map { (error, errorNum) =>
+            s"AtLeastOneEntryWithSeriesAndNullParentError $errorNum:" ->
+              error.swap.toOption.map(nel => InvalidValidationResult(nel.toList)).get
+          }
+      }.toMap
+
+      entriesThatFailedValidation = entriesCompletelyValidated.flatMap { case (_, entries) =>
+        entries.collect {
+          case entry if entry.exists { case (_, value) => value.isInvalid } =>
+            entry.map { case (entryName, value) =>
+              val transformedValue: ValidValidationResult | InvalidValidationResult =
+                value match {
+                  case Validated.Valid(value)                  => ValidValidationResult(value.toString)
+                  case Validated.Invalid(nonEmptyListOfErrors) => InvalidValidationResult(nonEmptyListOfErrors.toList)
+                }
+              entryName -> transformedValue
+            }
+        }
+      }.toList
+
+      (numOfJsonStructureErrors, numOfEntriesWithFailures) = (jsonStructuralErrors.size, entriesThatFailedValidation.length)
+      _ <- IO.whenA((numOfJsonStructureErrors + numOfEntriesWithFailures) > 0) {
+        lazy val keyWithoutOriginalFileName = key.split('/').dropRight(1).mkString("/")
+        lazy val keyOfResultsFile = s"$keyWithoutOriginalFileName/metadata-entries-with-errors.json"
+        lazy val fullLocationOfResultsFile = s"${bucket}/$keyOfResultsFile"
+        val jsonStructureErrorMessage =
+          if numOfJsonStructureErrors > 0 then
+            val s = if numOfJsonStructureErrors == 1 then "" else "s"
+            s"$numOfJsonStructureErrors thing$s wrong with the structure of the metadata.json for batchId '${input.batchId}'; the results can be found here: $fullLocationOfResultsFile\n\n"
+          else ""
+
+        val entriesErrorMessage =
+          if numOfEntriesWithFailures > 0 then
+            val (entryOrEntries, hasOrHave) = if numOfEntriesWithFailures == 1 then ("entry", "has") else ("entries", "have")
+            s"$numOfEntriesWithFailures $entryOrEntries (objects) in the metadata.json for batchId '${input.batchId}' $hasOrHave failed validation; the results can be found here: $fullLocationOfResultsFile"
```

Comment:
I've just noticed you are doing something similar to this further down where you have `mapToSendAsJson` 
You could change that so it sends the map in a similar way.

Also, if you're sending a map as json, why do we need the verbose logs?

Reply:
Ah, ok cool, that looks good.

So, how it is now, it posts the first message (the verbose one), just once to inform you that there is something wrong with the structure/entries, the `batchId` and where to find the file for more information. It's almost like a summary/title and something that you can search for if the there's a lot in the logs; I wanted it to look like an Exception message.

Then, only for entities, it will log the `id` of the entry and the field names that have an error, so that you can look them up in the results file because I'm thinking of the case where you have 100s/1000s of entries and only a subset have issues, so it's nice to be able to take the `id` (from the json log message) and search for it and then only look at the fields specified